File: flutter_module/part2/completed/DogBreeds/my_flutter_module/lib/DogBreed.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:my_flutter_module/SizeMetric.dart';

part 'DogBreed.g.dart';

@JsonSerializable()
class DogBreed {
  int id;
  String name;
  SizeMetric weight;
  SizeMetric height;
  String? description;
  String? bred_for;
  String? breed_group;
  String life_span;
  String? temperament;
  String? reference_image_id;

  DogBreed(
      this.id,
      this.name,
      this.weight,
      this.height,
      this.description,
      this.bred_for,
      this.breed_group,
      this.life_span,
      this.temperament,
      this.reference_image_id);

  factory DogBreed.fromJson(Map<String, dynamic> json) =>
      _$DogBreedFromJson(json);
  Map<String, dynamic> toJson() => _$DogBreedToJson(this);
}

File: flutter_module/part2/completed/DogBreeds/DogBreeds/DogBreed.swift
//
//  DogBreed.swift
//  DogBreeds
//

import Foundation

struct DogBreed: Codable, Identifiable {
    struct Weight: Codable {
        let imperial: String
        let metric: String
    }

    struct Height: Codable {
        let imperial: String
        let metric: String
    }

    let id: Int
    let name: String
    let weight: Weight
    let height: Height
    let description: String?
    let bred_for: String?
    let breed_group: String?
    let life_span: String
    let temperament: String?
    let reference_image_id: String?
}
